src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAYLIST_SIZE = 15;
+    private int listSize;
+    private int lastRecord = 0;
+
+    private T[] listArray = (T[])(new Object[DEFAULT_ARRAYLIST_SIZE]);
+
+    public ArrayList() {
+        listSize = DEFAULT_ARRAYLIST_SIZE;
+    }
+
+    private void growArray(int records) {
+        T[] tempArray = (T[])(new Object[listArray.length]);
+        for (int i = 0; i < listArray.length; i++) {
+            tempArray[i] = listArray[i];
+        }
+        listArray = (T[])(new Object[listArray.length + records]);
+        for (int i = 0; i < tempArray.length; i++) {
+            listArray[i] = tempArray[i];
+        }
```

Comment:
![image](https://user-images.githubusercontent.com/81781479/184500947-359d48a4-8927-4381-b4f5-ebc0d45d2605.png)
Before you create PR with your homework, you need to go through the description of the task and checklist under the task, and correct all the points described there. The mentor will not check the work until the checklist items are corrected